pkg/readiness/ready_tracker.go

```diff
@@ -89,6 +89,17 @@ type Tracker struct {
 	trackListerPredicateOverride retryPredicate
 }
 
+type WrapFakeClientWithMutex struct {
```

Comment:
This is pure test code, it should live in the test package.

Reply:
Updated. moved code to test file